### PR TITLE
move name inferrer inside destroyCommand.destroy

### PR DESCRIPTION
**Pull request by teresaromero**
# Proposed changes

Fixes https://okteto.atlassian.net/browse/DEV-706

Debugged the issue and found that the `destroy` command was not taking into account the manifest.name option when is run inside a deploy manifest.

The name inferrer was setting the name as the repo name when the configmap was not found, as a fallback. After this was set, when the manifest was read, the name inside the manifest was not taken into account.

Proposed changes:
- create a buildControlInterface to be able to use the correct name when providing the build controller
- decide the ops.Name in the same place once we have the manifest to have all params taken into account

## How to validate

Build the binary and deploy a dev instance. Reproduce the steps as mentioned on the ticket. 
Fork with changes without the installation of 2.31 https://github.com/teresaromero/okteto-test 

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

**Review by ifbyol — APPROVED** [review]
For the future, we will need to unify all this logic for all the commands. It should be identical in most of the cases, but we have duplicated a lot of this code and it generates this kind oof issues (a command doesn't work as the others)

I was testing it and it works fine. My comment inline is more about double-checking that moving the logic to calculate the proper name doesn't affect to things that are in the middle of where it was before and it is now

File: cmd/destroy/destroy.go [hunk 1]
```diff
@@ -364,15 +353,36 @@ func (dc *destroyCommand) destroy(ctx context.Context, opts *Options) error {
 	}
 
 	opts.Manifest = manifest
+
+	// name for cfg map being destroyed
+	// name set by flag has priority over manifest name
+	// if no name is set, the name of the manifest is used
+	// if no name is set in the manifest, the name is inferred
+	if opts.Name == "" && opts.Manifest.Name != "" {
```
[the remark this answers is not in the document]
Comment (teresaromero) [reply cmd/destroy/destroy.go]: you are right. i did not see the kubeconfig read of the name... the vision i have is to unify where the name and manifest.name are set so the read operations are done after...

i've updated with a commit to move this before reading the name for anything... this involves reading the manifest earlier in the destroy cmd. I've been testing it and i dont see any issue now. With the previous change the name could be not inferred.

File: cmd/destroy/destroy.go [hunk 2]
```diff
@@ -189,23 +193,6 @@ If you need to destroy external resources (like s3 buckets or other Cloud resour
 				return oktetoErrors.ErrContextIsNotOktetoCluster
 			}
 
-			cwd, err := os.Getwd()
-			if err != nil {
-				return fmt.Errorf("failed to get the current working directory: %w", err)
-			}
```
Comment (ifbyol) [inline cmd/destroy/destroy.go:195]: You don't have to remove this. In line 172, there might be a change in the working directory of the command, so `initialWd` might not be the correct one you have to pass to `setOptionsNameAndManifestName`

**Review by andreafalzetti — APPROVED** [review]

File: cmd/destroy/destroy.go (hunk 2 above)
Comment (teresaromero) [reply cmd/destroy/destroy.go]: reverted this change

**Pull request merged**